### PR TITLE
Messages API additions

### PR DESCRIPTION
**Pull request by rowanseymour**
* Searching by text
* Reversing order of results

### CODE REVIEW

File: temba/api/views.py [hunk 1]
```diff
@@ -762,7 +766,10 @@ def get_queryset(self):
         if broadcasts:
             queryset = queryset.filter(broadcast__in=broadcasts)
 
-        return queryset.order_by('-created_on').select_related('labels')
+        reverse_order = self.request.QUERY_PARAMS.get('reverse', None)
```
Comment (rowanseymour) [inline temba/api/views.py]: You'd switch the results ordering to oldest first by including `reverse=1` or `reverse=true` etc

File: temba/api/views.py [hunk 2]
```diff
@@ -2571,11 +2580,10 @@ def get_queryset(self):
...
 
         archived = self.request.QUERY_PARAMS.get('archived', None)
-        if not archived is None:
-            archived = True if archived.lower() in ['true', 'y', 'yes', '1'] else False
-            queryset = queryset.filter(is_archived=archived)
+        if archived is not None:
+            queryset = queryset.filter(is_archived=str_to_bool(archived))
 
-        return queryset.select_related('label')
+        return queryset.prefetch_related('labels')
```
Comment (rowanseymour) [inline temba/api/views.py]: `select_related` doesn't work for M2M fields.. also label isn't a field

File: temba/api/views.py [hunk 3]
```diff
@@ -620,7 +620,7 @@ class MessagesEndpoint(generics.ListAPIView):
       * **group_uuids** - the UUIDs of any groups the contact belongs to (string) (filterable: ```group_uuids``` repeatable)
       * **direction** - the direction of the SMS, either ```I``` for incoming messages or ```O``` for outgoing (string) (filterable: ```direction``` repeatable)
       * **labels** - Any labels set on this message (filterable: ```label``` repeatable)
-      * **text** - the text of the message received, not this is the logical view, this message may have been received as multiple text messages (string)
+      * **text** - the text of the message received, not this is the logical view, this message may have been received as multiple text messages (string) (filterable: ```text```)
```
Comment (nicpottier) [inline temba/api/views.py]: not yours, but correct 'not' to 'note'

Comment (nicpottier) [inline temba/api/views.py]: Also terrifying, let's not document filtering by this.

**Comment by nicpottier** [comment]
Looks good to me otherwise.

File: temba/api/views.py (hunk 3 above)
Comment (ericnewcomer) [inline temba/api/views.py]: +1 on leaving text search via api undocumented

**Pull request merged**